pyproject.toml

```diff
@@ -75,6 +75,7 @@ dependencies = [
     "botorch",
     "pyro-ppl",
     "gpytorch",
+    "omegaconf",
```

Comment:
Re:versions, I think the "Recursion way" has been to be as unspecific as possible in `pyproject` but to update the `requirements/*` files every month, and then if something breaks with latest version to pin it in `pyproject`.

Reply:
We have been following the same strategy (just using conda instead of pip). In short, our "library `env.yml` files" contain _mostly_ lower bounds or no bounds and for any specific project (internal, research, client, etc), we were using specific `env.yml` files with more specific bounds (including equal and upper bounds).

I think it's a nice setup, so you can quickly catch any incompat with the latest versions of your upstream deps. On that point, be aware that it's super easy to schedule a weekly CI run for your tests, so even when not working on a lib you know whether it's still compatible with the latest upstream versions. See this as an example: https://github.com/datamol-io/datamol/blob/fc0431b0d62da3d160338f887649374f51c35d22/.github/workflows/test.yml#L11-L12

Reply:
Noted, as hadim said, we use similar principles, my comment was more about lower bounds, since in `omegaconf`,   `gpytorch` and `botorch` often have breaking changes.